### PR TITLE
Refactor ListObjectsV1 and V2

### PR DESCRIPTION
**Pull request by masterSplinter01**
Separate ListObject for different versions.
Remove useless grouping of keys on V2.

### CODE REVIEW

File: api/handler/object_list.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+			Size:         obj.Size,
+			LastModified: obj.Created.Format(time.RFC3339),
+
+			Owner: Owner{
+				ID:          obj.Owner.String(),
+				DisplayName: obj.Owner.String(),
+			},
+
+			ETag: obj.HashSum,
+			// StorageClass: "",
```
Comment (alexvanin) [inline api/handler/object_list.go:65]: Remove if unused.

File: api/handler/object_list.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
```
Comment (alexvanin) [inline api/handler/object_list.go:128]: Remove if unused.

**Review by alexvanin — APPROVED** [review]
Make sure to figure out why amount of passed s3 compatibility tests decreased. 
I think @KirillovDenis noticed that it doesn't related to dirs as an object, so I wonder what else influenced the tests. 

Maybe it is related to `api.WriteErrorResponse(r.Context(), w, err, r.URL)` in `registerAndSendError`.

But overall, LGTM.

File: api/handler/object_list.go (hunk 1 above)
Comment (masterSplinter01) [reply api/handler/object_list.go]: Removed

File: api/handler/object_list.go (hunk 2 above)
Comment (masterSplinter01) [reply api/handler/object_list.go]: Removed

**Comment by masterSplinter01** [comment]
I've fixed the issues with test.

File: api/handler/object_list.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+	// fill common prefixes
+	for i := range list.Prefixes {
+		res.CommonPrefixes = append(res.CommonPrefixes, CommonPrefix{
+			Prefix: s3PathEncode(list.Prefixes[i], p.Encode),
+		})
+	}
+
+	// fill contents
+	for _, obj := range list.Objects {
```
Comment (KirillovDenis) [inline api/handler/object_list.go:53]: Can we separate this into a separate function and use in `encodeV2`?

Comment (masterSplinter01) [reply api/handler/object_list.go]: Moved filling prefixes and contents into separate functions.

File: api/handler/object_list.go [hunk 4]
```diff
@@ -188,6 +150,31 @@ func parseListObjectArgs(r *http.Request) (*layer.ListObjectsParamsCommon, error
...
 
+func fillPrefixes(dst *[]CommonPrefix, src []string, encode string) {
+	for _, obj := range src {
+		*dst = append(*dst, CommonPrefix{
+			Prefix: s3PathEncode(obj, encode),
+		})
+	}
+}
+
+func fillContents(dst *[]Object, src []*layer.ObjectInfo, encode string) {
```
[the remark this answers is not in the document]
Comment (masterSplinter01) [reply api/handler/object_list.go]: I don't see any difference, but fixed.

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**